### DISCUSSION TITLE
Embed Remix in a server side application

### DISCUSSION THREAD

**Discussion by tomtomdurrant** [remix-run/remix#3347] [Q&A]
Hi,

We're evaluating Remix as a potential framework for a service we're building at my work. The long and short of it is that we have a PHP application serving a header and a footer and that we would like the Remix app to be the main content that is served in the middle of it. Because the company I work for is in e-commerce they have requested that the app is SEO crawlable and all the good stuff that comes with server rendering a React application. Therefore they didn't like the idea of putting the Remix app in an iFrame and calling it a day.

Initially I thought that we could request the HTML from the corresponding Remix route and then proxy the subsequent requests (chunk.js etc) back to the Remix application. However this seems to duplicate the HTML. E.g. if all you have is a button on the page, when the JS comes back and hydrates the DOM, you'll then end up with 2 buttons. Once from the initial HTML and once from the hydration.

Can anyone think of what might be the best way of doing this? Basically "embedding" a Remix application inside another, totally separate PHP application. 

I hope this makes sense and literally _any_ help would be much appreciated.

**Comment by machour** [comment]
Just throwing an idea here: what if you inverted the idea, developed the Remix app as the main app, and considered the header & footer rendered by PHP as data, that you would retrieve from your root loader ?

**Reply by tomtomdurrant** [reply]
We've considered _something_ like that, but not that specifically. We'll look into it, thanks! (I wish we could just re do the whole thing in remix, but the business are insistent we have to use the header and footer 🤷 )

**Reply by machour** [reply]
Let me know how it goes, I'm going to be migrating some PHP apps in the future, definitely interested in solving this kind of problems.
There's also a [discord server](https://discord.com/invite/remix) if you want to have a chat
